hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/BlockManagerImpl.java

```diff
@@ -312,6 +325,105 @@ public List<BlockData> listBlock(Container container, long startLocalID, int
     }
   }
 
+  public Set<MissingBlock> headBlocks(Container container, Set<Long> localIds) throws IOException {
+    Preconditions.checkNotNull(container, "container cannot be null");
+    Preconditions.checkNotNull(localIds, "local ID cannot be null");
+    if (localIds.isEmpty()) {
+      return new HashSet<>();
+    }
+
+    container.readLock();
+    try {
+      KeyValueContainerData containerData = (KeyValueContainerData) container.getContainerData();
+      if (!containerData.getLayoutVersion().equals(ContainerLayoutVersion.FILE_PER_BLOCK)) {
+        throw new UnsupportedEncodingException("Not support Container Layout " + containerData.getLayoutVersion());
```

Comment:
I'm not sure this is the correct type to use. The Java doc says this is for character encodings. We probably want to throw some type of `StorageContainerException`.

Reply:
Corrected to `UnsupportedOperationException`